website/api/bonsai/bonsai-on-eth.md

```diff
@@ -2,18 +2,133 @@
 
 _Warning: Bonsai is still in early development. Do not use in production._
 
-Bonsai can be used as a [zk coprocessor] for Ethereum, allowing you to request and receive [verified proofs] directly from your smart contract.
+Bonsai can be used as a [zk coprocessor] for Ethereum, allowing you to request [verified proofs] via an off-chain REST API interface or on-chain directly from your smart contract.
 
 At a high level, here's how it works:
 
-![Bonsai ETH Relay overview](/img/eth-relay-diagram.jpg)
+![Bonsai ETH Relay overview](/img/bonsai_ethereum.png)
 
-<!-- TODO: Highlight the relay sections of this diagram -->
+1. Users can delegate their smart contract's logic to Bonsai via a `Request Callback` interface. This interface is accessible both _off-chain_ through the HTTP REST API of the `Bonsai Relayer` and _on-chain_ via the [Bonsai Relay Contract].
+2. The `Bonsai Relayer` sends the proof request to Bonsai.
+3. Bonsai generates a [Groth16 SNARK proof] and its result, encapsulated in a journal.
+4. The `Bonsai Relayer` submits this proof and journal on-chain to the [Bonsai Relay Contract] for validation.
+5. If validated, the journal is dispatched to the user's smart contract via the specified callback.
 
-Our Bonsai-ETH Relay acts as a middle-man between your app contract and the Bonsai proving service. In order to use the Bonsai ETH Relay, dApp developers must have completed the following steps:
+To integrate your application with Bonsai, you'll need to:
 
-1. Deployed a [smart contract] on chain that calls the relay contract
-2. Uploaded a compiled [program] for the zkVM to the Bonsai proving service
+1. Write the [program] you want proven and upload it to Bonsai
+2. Write the on-chain part of your application and deploy it
+
+## Interfaces
+
+The Bonsai Ethereum Relay provides both an _off-chain_ (vie the HTTP REST API of the `Bonsai Relayer`) and an _on-chain_ (via the [Bonsai Relay Contract]) interface to send `Callback requests`.
+
+### Off-chain
```

Comment:
The amount of duplicated code and writing between here an the readme worries me a little. If it's possible to de-duplicate this by having one reference the other it would probably reduce maintenance needs.

Reply:
I decided to keep the more information we can on the docs and streamline the README